cfme/services/catalogs/catalog.py

```diff
@@ -1,115 +1,110 @@
 
 
-cfg_btn = partial(tb.select, "Configuration")
-catalog_tree = partial(accordion.tree, "Catalogs")
+from . import ServicesCatalogView
+
+
+class CatalogForm(ServicesCatalogView):
+    title = Text('#explorer_title_text')
+
+    name = Input(name='name')
+    description = Input(name="description")
+
+    save_button = Button('Save')
+    cancel_button = Button('Cancel')
+
+
+class CatalogsView(ServicesCatalogView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.in_explorer and
+            self.title.text == 'All Catalogs' and
+            self.catalogs.is_opened and
+            self.catalogs.tree.currently_selected == ["All Catalogs"])
+
+
+class AddCatalogView(CatalogForm):
 
-listview_table = CheckboxTable(table_locator='//div[@id="list_grid"]/table')
+    add_button = Button("Add")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Adding a new Catalog" and
+            self.catalogs.is_opened
+        )
 
-item_multiselect = web_ui.MultiSelect(
-    "//select[@id='available_fields']",
-    "//select[@id='selected_fields']",
-    "//div[@id='column_lists']//a[contains(@href, 'button=right')]/img",
-    "//div[@id='column_lists']//a[contains(@href, 'button=left')]/img")
 
-form = web_ui.Form(
-    fields=[('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('button_multiselect', item_multiselect),
-            ('add_button', form_buttons.add),
-            ('save_button', form_buttons.save),
-            ('cancel_button', form_buttons.cancel)])
+class EditCatalogView(CatalogForm):
 
-item_form = web_ui.Form(
-    fields=[('type_select', "//select[@id='st_prov_type']"),
-            ('name_text', Input("name")),
-            ('description_text', Input("description")),
-            ('display_checkbox', Input("display")),
-            ('add_button', form_buttons.add)])
+    save_button = Button('Save')
+    reset_button = Button('Reset')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Editing Catalog{}".format(self.obj.name) and
+            self.catalogs.is_opened
+        )
 
-match_page = partial(match_location, controller='catalog', title='Catalogs')
 
+class DetailsCatalogView(ServicesCatalogView):
 
-def _all_catalogs_add_new(_):
-    if BZ(1213863).blocks:
-        sel.pytest.skip("Blocker on the bug 1213863")
-    catalog_tree('All Catalogs')
-    cfg_btn('Add a New Catalog')
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == 'Catalog "{}"'.format(self.obj.name) and
+            self.catalogs.is_opened
+        )
 
 
 class Catalog(Updateable, Pretty, Navigatable):
-    """Represents a Catalog"""
-    pretty_attrs = ['name', 'items']
 
     def __init__(self, name=None, description=None, items=None, appliance=None):
+        Navigatable.__init__(self, appliance)
```

Comment:
Can we keep the keyword reference here? `Navigatable.__init__(self, appliance=appliance)`